service/controller/v7/resource/encryptionkey/delete.go

```diff
@@ -15,14 +16,18 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", "deleting encryptionkey secret")
-
-	err = r.k8sClient.CoreV1().Secrets(key.CertificateEncryptionNamespace).Delete(key.CertificateEncryptionSecretName(customObject), &metav1.DeleteOptions{})
-	if err != nil {
-		return microerror.Mask(err)
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting encryptionkey secret upon delete event")
+
+		err = r.k8sClient.CoreV1().Secrets(key.CertificateEncryptionNamespace).Delete(key.CertificateEncryptionSecretName(customObject), &metav1.DeleteOptions{})
+		if errors.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "encryptionkey secret already deleted")
+		} else if err != nil {
+			return microerror.Mask(err)
+		} else {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "deleted encryptionkey secret upon delete event")
```

Comment:
I updated the logging here in the latest version. I tried to incorporate the "why" but this needs more feedback.

Reply:
I guess this is fine. This is relatively simple implementation after all and there are no third parties (i.e. other services such as cloud provider etc.) involved here.